### PR TITLE
DOC: allclose doesn't require matching shapes

### PR DESCRIPTION
**Pull request by spencerahill**
Closes https://github.com/numpy/numpy/issues/5654

Happy to iterate on this...first time numpy contributor but thought this was straightforward enough to take a first stab at myself.  Thanks in advance.

### CODE REVIEW

File: numpy/core/numeric.py [hunk 1]
```diff
@@ -2448,6 +2448,10 @@ def allclose(a, b, rtol=1.e-5, atol=1.e-8, equal_nan=False):
     The above equation is not symmetric in `a` and `b`, so that
     `allclose(a, b)` might be different from `allclose(b, a)` in
     some rare cases.
+        
+    The comparison of `a` and `b` includes standard broadcasting,
+    which means that `a` and `b` need not have the same shape in order for
+    allclose(a, b) to evaluate to True.  (The same is true for `np.equals`.)
```
Comment (eric-wieser) [inline numpy/core/numeric.py]: I think omitting the `np.` will make this link to `equals` in the docs, but I'm not sure.

Comment (spencerahill) [reply numpy/core/numeric.py]: Ok, I'll omit the `np.`.  Would it make sense to also add `equals` under See Also?  Seems to me yes.

File: numpy/core/numeric.py [hunk 2]
```diff
@@ -2448,6 +2448,10 @@ def allclose(a, b, rtol=1.e-5, atol=1.e-8, equal_nan=False):
     The above equation is not symmetric in `a` and `b`, so that
     `allclose(a, b)` might be different from `allclose(b, a)` in
```
Comment (charris) [inline numpy/core/numeric.py]: Hmm, should be double backticks around `allclose(a, b)`, same for `allclose(b, a)`.  Not yours, but could you fix it?

File: numpy/core/numeric.py [hunk 3]
```diff
@@ -2448,6 +2448,10 @@ def allclose(a, b, rtol=1.e-5, atol=1.e-8, equal_nan=False):
...
     some rare cases.
+        
+    The comparison of `a` and `b` includes standard broadcasting,
```
Comment (charris) [inline numpy/core/numeric.py]: includes  <- uses

File: numpy/core/numeric.py [hunk 4]
```diff
@@ -2448,6 +2448,10 @@ def allclose(a, b, rtol=1.e-5, atol=1.e-8, equal_nan=False):
...
+    which means that `a` and `b` need not have the same shape in order for
+    allclose(a, b) to evaluate to True.  (The same is true for `equals`.)
```
Comment (charris) [inline numpy/core/numeric.py]: Double backticks around `allclose(a, b)`. No parenthesis around the closing sentence.

**Comment by charris** [comment]
If you put `[ci skip]` in the commit message test won't be run. They are not needed for documentation.

**Comment by spencerahill** [comment]
@charris thanks for the review.  Requests implemented.

File: numpy/core/numeric.py [hunk 5]
```diff
@@ -2446,9 +2446,14 @@ def allclose(a, b, rtol=1.e-5, atol=1.e-8, equal_nan=False):
...
 
     The above equation is not symmetric in `a` and `b`, so that
-    `allclose(a, b)` might be different from `allclose(b, a)` in
+    ``allclose(a, b)`` might be different from ``allclose(b, a)`` in
     some rare cases.
 
+    The comparison of `a` and `b` uses standard broadcasting, which
+    means that `a` and `b` need not have the same shape in order for
+    ``allclose(a, b)`` to evaluate to True.  The same is true for
+    ``equal`` but not ``array_equal``.
```
Comment (charris) [inline numpy/core/numeric.py]: The equal and array_equal can use single backticks. The difference is that the single backticks denote links, whereas the double backticks denote code.

**Comment by charris** [comment]
Fixed and merged. Thanks @spencerahill .

**Comment by spencerahill** [comment]
My pleasure.  Thank you @charris and @eric-wieser for your help.

**Pull request merged**